Newliz/branch

ejporter66 opened the pull request:
delete books page check comments prev PR

ejporter66 committed:
made changes too addbooks based on comments

ejporter66 committed:
add DeleteBooks function

added a delete books function in the admin grid. employees can search books by title and author and delete them in the database via a checkbox and click button. Heads up I think bestSellers is the only collection being fetched. Anyone is welcome to troubleshoot.

ejporter66 committed:
pulled main's latest repo

had to pull from main and get up to date and re did pull request

ejporter66 committed:
cleaned it up and improved function

all collections are pulling into the search query now along with title, description, and thumbnail if available

Mark32582 commented on src/components/DeleteBooks/DeleteBooks.js:
```diff
@@ -0,0 +1,194 @@
+
+      <ul>
+        {selectedCollections.map((collectionName) => (
+          <li key={collectionName}>
+            <button onClick={() => fetchBooks(collectionName)}>
+              {collectionName}
+            </button>
+          </li>
```
why dont we set these to radio buttons instead of standard buttons that way you can see what is selected and which category your searching

Mark32582 commented on src/components/AddBooks/AddBook.js:
```diff
@@ -3,29 +3,37 @@ import axios from "axios";
   const [quantity, setQuantity] = useState("");
-  const [search, setSearch] = useState("");
-  const [books, setBooks] = useState([]);
+  const [search, setSearch] = useState(propSearch);
+  const [books, setBooks] = useState(propBooks);
```
these are global states we don't need to override them here please delete both these lines and just use the props above. Since we only want these results here lets give these a different name search could be orderSearch and books could be orderBook that way the search only in this area

Mark32582 commented on src/components/AddBooks/AddBook.js:
```diff
@@ -3,29 +3,37 @@ import axios from "axios";
+    setEmployee,
+    users,
+    setUsers,
+    search: propSearch,
+    setSearch: propSetSearch,
+    books: propBooks,
+    setBooks: propSetBooks,
```
we don't need to change these they are global states that work in multiple areas having them overwritten here doesn't allow the site to function properly

Mark32582 commented on src/components/AddBooks/AddBook.js:
```diff
@@ -80,71 +99,74 @@ const AddBooks = () => {
-          onChange={handleQuantityChange}
+          onChange = {(e) => setQuantity(e.target.value)}
           placeholder="Enter quantity"
         />
```
why dont we add radio buttons here to choose what collection we want to add the book to

Mark32582 commented on src/components/AddBooks/AddBook.js:
when you bring in the navigation you need to add the props to it look at homepage.js for an example. also you need to bring in both logon and cart and place them directly below the navigation component

Mark32582 commented on src/components/DeleteBooks/DeleteBooks.js:
```diff
@@ -0,0 +1,194 @@
+        setEmployee={setEmployee}
+        users={users}
+        setUsers={setUsers}
+      />
```
Dont forget to add Cart here as well with its props

Mark32582 reviewed (changes requested):
a few minor changed and updates and we can merge this in looks great  sorry also missed on your map function in addBooks you need to add a ? in front of the . between books and map   `books?.map(`

ejporter66 committed:
modified based on comments

modified both of the files based on the comments. I believe I did everything you mentioned minus the radio buttons on the add books page. I need to figure out how to reverse the logic in the delete section.

Mark32582 reviewed (changes requested):
Having routing issues where when you click the buttons it doesn't pull up the pages please address this also before moving on please merge main into your branch to get all the other updates we have working to make sure there are no conflicts

ejporter66 committed:
modifications deletebooks

adjusted routing

ejporter66 committed:
modifications deletebooks

adjusted routing

ejporter66 committed:
same mod update
